sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1869,31 +1869,32 @@ void ProgramManager::addImages(sycl_device_binaries DeviceBinary) {
     for (sycl_offload_entry EntriesIt = EntriesB; EntriesIt != EntriesE;
```

Comment:
will the iterator work when the input binary is using legacy entry format? sizeof entries are different.